src/AddUserScreen.tsx

```diff
@@ -12,9 +16,55 @@ export default function newUser() {
   const todayDate = new Date();
   const maxDate = todayDate.toISOString().split('T')[0];
 
-  function handleSubmit(e: any) {
+  const history = useHistory();
+  const [userSubmission, { loading }] = useMutation(CREATE_USER);
+
+  if (loading) {
+    return (
+      <div className='App'>
+        <header className='App-header'>
+          <ClipLoader />
+        </header>
+      </div>
+    );
+  }
+
+  const userMutation = async (
+    name: string,
+    email: string,
+    password: string,
+    phone: string,
+    birthDate: string,
+    role: string,
+  ) => {
+    if (newUserValidation(name, email, password, phone, birthDate, role)) {
+
+      try {
+        await userSubmission({
+          variables: {
+            data: {
+              name: name,
+              email: email,
+              phone: phone,
+              birthDate: birthDate,
+              password: password,
+              role: role
+            }
+          }
+        });
```

Comment:
se vc passar o objeto igual eu falei no ultimo comentario aqui vc poderia deixar somente o objeto passando para o data

```suggestion
        await userSubmission({
          variables: {
            data
          }
        });
```